terraform/maas/lb.tf

```diff
@@ -0,0 +1,89 @@
+resource "aws_lb" "maas_regional" {
+  name               = "maas-load-balancer"
+  internal           = true
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.maas_lb_sg.id]
+  subnets            = data.aws_subnet_ids.public_subnets.ids
+
+  enable_deletion_protection = false
+  tags = merge(
+    local.common_tags,
+    map(
+      "Name", "maas-load-balancer"
+    )
+  )
+}
+
+resource "aws_security_group" "maas_lb_sg" {
+  name        = "maas-lb-sg"
+  description = "Allow maas https traffic into load balancer"
+  vpc_id      = join(", ", data.aws_vpcs.moz_internal_us_west_2.ids)
+
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["10.49.0.0/16", "10.51.0.0/16"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(
+    local.common_tags,
+    map(
+      "Name", "maas-lb-sg"
+    )
+  )
+}
+
+resource "aws_lb_target_group" "maas_lb_target_group" {
+  name        = "maas-lb-tg"
+  port        = 5240
```

Comment:
can we make the port a variable? Maybe we'll never change it, but it'd be nice to see it in the vars for reference and to not have it hard-coded (the 443 seems different to me as a standard port).

Reply:
We can set this to a variable, I don't have a problem with that.  Although, I don't think this port will change since it is a standard for maas API/UI.